integration_tests/sdk/aqueduct_tests/param_test.py

```diff
@@ -625,8 +625,15 @@ def append_row_to_df(df, row):
         output_artifact_list.append(output)
         input_data_list.append(input_df)
 
+    with pytest.raises(
+        InvalidUserActionException,
+        match="Cannot create a flow with local data."
+    ):
+        flow = publish_flow_test(
+            client, artifacts=output_artifact_list, name=flow_name(), engine=engine
```

Comment:
Let's use `client.publish_flow()` if we expect the flow to error before sending anything to the server.